src/app/team/serviceForm/whoIsThisClothingFor/ServiceWhoIsThisClothingFor.jsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { compose, withProps } from 'recompose';
+import { getService, getServiceWearerAge, getServiceId } from '../../../../selectors/service';
+import { selectLocationError } from '../../../../selectors/location';
+import * as actions from '../../../../actions';
+import { Form } from '../../../../components/form';
+import WhatKindOfClothingEdit from './WhatKindOfClothingEdit';
+import WhatKindOfClothingView from './WhatKindOfClothingView';
+
+const FormComponent = compose(withProps({
+  ViewComponent: WhatKindOfClothingView,
+  EditComponent: WhatKindOfClothingEdit
+}))(props => <Form {...props} />);
+
+const mapStateToProps = (state, ownProps) => ({
+  resourceData: getService(state, ownProps),
+  value: getServiceWearerAge(state, ownProps),
+  id: getServiceId(ownProps),
+  resourceLoadError: selectLocationError(state, ownProps),
+});
+
+const mapDispatchToProps = (dispatch, ownProps) => ({
+  fetchResourceData: bindActionCreators(actions.getLocation, dispatch),
+  updateValue: (wearerAge, serviceId, metaDataSection, fieldName) =>
+    dispatch(actions.updateService({
+      locationId: ownProps.match.params.locationId,
+      serviceId,
+      params: { wearerAge },
```

Comment:
Hmm... this may not be clear from the designs, but I think wearerAge isn't necessarily the only parameter being set here. Looking at the existing data in our DB, at the Airtable I shared a while back, or at the way our filters work right now, you can see that wearerAge actually only supports the values "adults" and "children", whereas the difference between men vs. women or boys vs. girls is currently captured by the "gender" eligibility parameter.
Of course, this is problematic, because it doesn't actually make it possible to capture the _connection_ between age and gender (e.g. there's no way to capture that a place has clothes only for women and boys).

What I'd suggest for now, so that you don't have to change the filters, but also don't have to make any schema changes, is this:
* Keep passing the values as you are right now, for "men", "women", "girls", and "boys".
* When you pass "men" and/or "women", also _add_ the value "adults". And when you pass "boys" and/or "girls", also add the value "children".
* Finally, when "men" is one of the values, also pass the "gender" eligibility parameter with "male", and when "women" is one of the values, pass "female".

That way, we're capturing all the granular data, but we're also backwards-compatible for the values expected by the filters.

Reply:
Great implementation.